### DISCUSSION TITLE
Strange unicode characters in rendered HTML

### DISCUSSION THREAD

**Discussion by mbforbes** [resend/react-email#1517] [Q&A] [answered]
One separate question — still involving rendering the "vercel-invite-user" example:
https://demo.react.email/preview/notifications/vercel-invite-user

Rendering using version 2.1.4 with:

```bash
yarn run email export --outDir out/ --pretty
```

My editor shows a ton of invisible unicode characters are generated after the preview and before the body.

![Screenshot 2024-06-24 at 3 24 11 PM](https://github.com/resend/react-email/assets/1170062/f07185d1-136e-4f76-b14b-1701d79a65ed)

In fact, it has trouble word-wrapping them, and they go way off to the right of the screen:

![Screenshot 2024-06-24 at 3 24 46 PM](https://github.com/resend/react-email/assets/1170062/5e654b0d-c561-4e7c-aa2e-91160528e669)

Is this intended? Or is it a bug?

**Comment by mbforbes** [comment, marked as answer]
I think the purpose of these non-rendering unicode characters is to create padding so that only the preview snippet is shown. (The preview snippet is rendered as as an otherwise-invisible `<div>` between the `<head>` and `<body>`.)

![Screenshot 2024-06-25 at 4 58 03 PM](https://github.com/resend/react-email/assets/1170062/92c5ac00-15d3-4a01-bf2c-08e31ac1b038)

^ example **top:** keeping the invisible unicode characters, **bottom:** deleting them.

this might be nice to add to the docs, perhaps https://react.email/docs/components/preview ?

**Reply by gabrielmfern** [reply]
That's correct, these characters are meant to do precisely that. It would indeed be interesting to have a mention of this in the docs.
